### PR TITLE
fix: setting of cache when timezone is involved

### PR DESCRIPTION
**Pull request by sshanzel**
Most of our streaks data are queried and the correct timezones are fetched. However, in CDC, the data being passed is in UTC, hence, in JS we need to convert it to the right timezone for a proper calculation.

Added to tests:
- Timezone > UTC
- Timezone < UTC

Validated the entire CDC process relating to the setting of cache, every end considers timezone now.

### CODE REVIEW

File: src/common/users.ts [hunk 1]
```diff
@@ -546,8 +546,9 @@ export const shouldAllowRestore = async (
 ) => {
   const { userId, lastViewAt: lastViewAtDb } = streak;
   const user = await con.getRepository(DbUser).findOneBy({ id: userId });
-  const today = new Date();
-  const lastView = new Date(lastViewAtDb);
+  const timezone = user.timezone || 'utc';
```
Comment (capJavert) [inline src/common/users.ts:549]: we have `DEFAULT_TIMEZONE` const probably better to use it

File: src/common/users.ts [hunk 2]
```diff
@@ -546,8 +546,9 @@ export const shouldAllowRestore = async (
...
+  const today = utcToZonedTime(new Date(), timezone);
+  const lastView = utcToZonedTime(new Date(lastViewAtDb), timezone);
```
Comment (capJavert) [inline src/common/users.ts:551]: `lastViewAtDb` is not `Date` already?

**Review by capJavert — APPROVED** [review]
Looks good, some questions mostly

File: src/common/users.ts (hunk 1 above)
Comment (sshanzel) [reply src/common/users.ts:549]: Oh, nice, yeah. Let's use that.

File: src/common/users.ts (hunk 2 above)
Comment (sshanzel) [reply src/common/users.ts:551]: No, this one came from CDC. So either a number or a string. You never know :hide_the_pain:

**Pull request merged**